### ISSUE TITLE
blastem_libretro waterbox? blastem_libretro.dll feature request? ALT core

### ISSUE THREAD

**Issue by Eggo-dotcom** [TASVideos/BizHawk#2771]
a feature request, and not replace but offer as well

blastem_libretro waterbox? blastem_libretro.dll feature request? ALT core?

**Comment by getCursorsExe** [comment]
Loading any savestate makes BizHawk vulnerable to crashes, so it needs to be updated if developers want to add it here...

**Comment by Eggo-dotcom** [comment]
Didnt realize it was broken, was just reading "its super accurate"

**Comment by getCursorsExe** [comment]
You don't need to close it, we need to wait for developers' comment.

**Comment by getCursorsExe** [comment]
It crashes most likely because it is a very new core.

**Comment by Eggo-dotcom** [comment]
Right, but my only reason for requesting it is hearing how much more accurate than the current core it is however if its prone to crashing I made a mistake requesting it

**Comment by YoshiRulz** [comment]
For those out of the loop:
> BlastEm is an open source, higly accurate emulator for the Genesis/Megadrive that runs on modest hardware. It is the first emulator to properly run Titan's impressive Overdrive 2 demo and it is the only other emulator besides Genesis Plus GX that can properly run the original Overdrive.

It's written in C with the usual SDL2. The source is on a self-hosted Mercurial repo and is licensed under the GPL v3+.

**Comment by Eggo-dotcom** [comment]
So then It would be a good idea to add as a second option core?

**Comment by getCursorsExe** [comment]
While it has many issues that crashes often (including savestates), it might be good idea but BizHawk devs need to fix these or the core will be completely unusable.

**Comment by Eggo-dotcom** [comment]
I was just thinking a stable core option would be a nice addition. Dont know how stable this is though if its truly prone to crashing

**Comment by getCursorsExe** [comment]
It is pretty stable if we don't use savestates.

**Comment by CasualPokePlayer** [comment]
> 
> 
> This core appears to be broken. Loading any savestate is the core's weak point and might crash BizHawk

Ok so waterbox it.

**Comment by Eggo-dotcom** [comment]
I thought it might give bizhawk a more accurate genesis core. If blastem is really that buggy i dont think its worth adding

**Comment by vadosnaprimer** [comment]
Aside from elaborate technical demos, which games does blastem emulate better than **upstream** gpgx?

**Comment by YoshiRulz** [comment]
Ah, when I copy-pasted that description from the website, I didn't notice the word "demo". BlastEm is actually [missing a lot of mappers and peripherals](https://www.retrodev.com/blastem/trac/query?version=0.5.0&type=completeness&report=11&order=priority) so there would be little benefit in adding it as an alternate core, unless it offered a substantial speed increase over GPGX in those games it does emulate.

**Comment by vadosnaprimer** [comment]
gpgx is also _really_ fast.

**Comment by Eggo-dotcom** [comment]
I belive the newest blastem 0.63 can run the full library. Its slower than genplgx but more accurate. That being said i dont think its stable enough. Too many ppl say it crashed. Using myself the win64 nightly no issues but when i tried the liberto version crash crash crash. So i cancel my request.

**Comment by vadosnaprimer** [comment]
> more accurate

Any game examples?

**Comment by getCursorsExe** [comment]
It is still a good idea, since it perfectly runs TiTAN Overdrive 2 Demo.

**Comment by vadosnaprimer** [comment]
You plan to TAS a demo?

**Comment by Eggo-dotcom** [comment]
> It is still a good idea, since it perfectly runs TiTAN Overdrive 2 Demo.

Runs the complete commercial library without hacks….

**Comment by getCursorsExe** [comment]
> You plan to TAS a demo?

No, but since it perfectly runs it, it means that this core is very accurate, pretty similar to hardware

**Comment by Eggo-dotcom** [comment]
> > You plan to TAS a demo?
> 
> No, but since it perfectly runs it, it means that this core is very accurate, pretty similar to hardware

Do what I did, switch to retroarch. Now I have both cores and sharers.

**Comment by nattthebear** [comment]
Doesn't look like a perfect core, but it'd be nice to have another Genesis in our pocket.

**Comment by getCursorsExe** [comment]
It has to be waterboxed, because the upstream one is unusable for TASing.

**Comment by getCursorsExe** [comment]
Compare with Genplus-gx: https://github.com/libretro/Genesis-Plus-GX/blob/master/core/state.c

**Comment by CasualPokePlayer** [comment]
Took a peek at doing this: going off it's front page:
> BlastEm depends on SDL2, GLEW and **OpenGL**.
OpenGL is an immediate no for waterboxing, that would need to be stripped out of the core, how easy that is I don't know.

There is another interesting aspect to this core: it seems to support dynarec. This wouldn't work in waterbox as it is (although it could in theory be made to work), but regardless seems like a NEW_CORE define disables this dynarec? Could potentially be interesting to look into, as I imagine that dynarec might be significantly faster compared gpgx (does libretro disable dynarec / the "it's slower" due to that?).

Regardless, it still doesn't look pretty to port over, SDL2 should probably be stripped out (does wbx even support that?). But if that's not so hard then it's probably painless?

**Comment by CasualPokePlayer** [comment]
https://github.com/CasualPokePlayer/BizHawk/tree/blastem/waterbox

Well, I took a stab at this. It builds. That's it. I have no idea if this even works. Probably does. Needs an actual interface and start c# side actually hooked up, but it doesn't seem so bad.

**Comment by getCursorsExe** [comment]
It doesn't even want do build for me.

**Comment by getCursorsExe** [comment]
No audio currently.

**Comment by CasualPokePlayer** [comment]
Yes because those were not hooked up. Stop complaining work in progresses are incomplete.

**Comment by getCursorsExe** [comment]
I'm just saying in case if you can't build it yet, but I managed to successfuly build it.

**Comment by CasualPokePlayer** [comment]
> EDIT: Got some interface in, it boots and plays games. A lot of shit needs to be hooked up. Not too bad ig

> it boots

**Comment by getCursorsExe** [comment]
I can't build it again. One of the files causes EVERY SINGLE PROJECT to fail compiling, preventing anything from loading.

**Comment by getCursorsExe** [comment]
Should we port from that fork to here?

**Comment by CasualPokePlayer** [comment]
Not everything is actually pushed. Stop trying to build incomplete crap

**Comment by getCursorsExe** [comment]
Okay.

**Comment by Shinobi-1987** [comment]
So this might hapen?

**Comment by YoshiRulz** [comment]
Anything "might happen", unless it's been closed as wontfix. But commenting isn't going to make it happen faster.

**Comment by Shinobi-1987** [comment]
> Anything "might happen", unless it's been closed as wontfix. But commenting isn't going to make it happen faster.

Wasnt looking to make it faster, hope i didnt come across rude.

**Comment by getCursorsExe** [comment]
The reason why I want you to fix building failures is because I'm really wanting to check for bugs. Having build failures makes it impossible for development to continue. Anyways, what commit in your branch broke this?
